# BreakPointSurveyor: Synthetic branch

*A comprehensive pipeline to generate, analyze, and visualize structural variants in synthetic datasets*

See [TCGA_Virus](https://github.com/ding-lab/BreakPointSurveyor) workflow for comprehensive documentation.

## Overview

The Synthetic branch generates inter- and intra-chromosomal translocations by concatenating reference segments
and creating synthetic (simulated) reads in those regions.  After realigning with BWA to a custom minimal reference,
the small BAMs are analyzed and visualized with a standard BPS workflow.
In addition, this workflow demonstrates gene and exon annotation functionality, including exon labels.

We consider two datasets:

### "Synthetic.9-22"
A breakpoint is generated by concatenating these two segments together:
* Segment 1: chr9:130,840,000-130,850,000
* Segment 2: chr22:23,260,000-23,270,000

### "IntraA"
Demonstrates intra-chromosomal analysis of a dataset of three concatenated segments `AC'B`, where
* Segment A : chr9:36833000,36843000 ( 10Kbp )
* Segment B : chr9:36843000,36853000( 10KBp, adjacent to A )
* Segment C : chr9:37843000,37853000 ( 10Kbp, some other region of chrom 9 )
    * Segment C' : reverse complement of Segment C


## Results

### "Synthetic.9-22"
Structure plot below illustrates discordant reads mapping to a junction between chr 9 and 22, together with copy number
in that region.

<img src="T_PlotStructure/plots/synthetic.9-22.AA.chr22_chr9.BreakpointSurvey.png" width="600"/>


### "IntraA"
<img src="T_PlotStructure/plots/IntraA.AA.chr22_chr9.BreakpointSurvey.png" width="600"/>

## Workflow
Stages associated with Synthetic workflow:

* **[A_Reference](A_Reference/README.md)**: Construct minimal reference consisting only of GRCh38 chromosomes 9 and 22.
* **[B_ExonGene](B_ExonGene/README.md)**: Define demonstration exons and genes.
* **[C_SyntheticBAM](C_SyntheticBAM/README.md)**: Construct a synthetic BAM to demonstrate simple inter-chromosomal breakpoint ("Synthetic.9-22")
* **[D_TestBAM](D_TestBAM/README.md)**: Construct a synthetic BAM to demonstrate simple intra-chromosomal breakpoint ("IntraA")
* **[F_Project](F_Project/README.md)**: Construct a master list of all BAMs
* **[G_Discordant](G_Discordant/README.md)**: Process realigned BAM file to extract both intra- and inter-chromosomal discordant reads.
* **[J_PlotList](J_PlotList/README.md)**: Generate PlotList to define regions for plotting.
* **[K_ReadDepth](K_ReadDepth/README.md)**: Evaluate read depth in target regions, and obtain BAM file statistics used for calculating copy number.
* **[N_DrawBreakpoint](N_DrawBreakpoint/README.md)** Draw breakpoints with coordinates given by posA, posB.
* **[O_DrawDepth](O_DrawDepth/README.md)** Create read depth/copy number panel GGP and add breakpoint predictions.
* **[P_DrawAnnotation](P_DrawAnnotation/README.md)** Create annotation panel GGP showing genes and exons
* **[Q_DrawHistogram](Q_DrawHistogram/README.md)**: Create histogram panel GGP showing distribution of read depth
* **[T_PlotStructure](T_PlotStructure/README.md)**: Assemble GGP panels into BPS structure plot and save as PDF and PNG files

## Installation

Installation instructions [available here](https://github.com/ding-lab/BreakPointSurveyor/blob/master/INSTALL.md).  Besides the core libraries, also install,
* BWA
* samtools
* Perl
  * Install required library with, `cpanm XML::XPath`

## Performance

* Downloading and processing of two-chromosome reference (A_Reference) is about 5 minutes, uses about 684M
* Total execution time for steps B-T about 25 seconds

