# BPS.TCGA_Virus.Lite - Synthetic branch
Sample BreakPoint Surveyor project.  Data for synthetic human-human breakpoint



## Getting Started
### Prerequisites

#### BreakPointSurveyor-Core
Install [BreakPointSurveyor-Core](https://github.com/ding-lab/BreakPointSurveyor-Core)

```git clone https://github.com/ding-lab/BreakPointSurveyor-Core```

#### Tigra-SV

#### Perl XML 

Used currently only for testing of sequence download.

```cpanm XML::XPath```

#### BWA

https://sourceforge.net/projects/bio-bwa/files/

### Installing
* Install from [GitHub](https://github.com/)

```
git clone https://github.com/ding-lab/BPS.TCGA_Virus.Lite
```

* Create and edit local configuration file

```cp bps.config.template bps.config```
  * Set `BPS_CORE` to location of BreakPointSurveyor-Core installation location
  * BPS_DATA points to data directory.  Defining `BPS_DATA=".."` (the default)
    will write the data from a given stage (e.g., A_Project) into a data directory 
    in that stage.  Alternatively, can define BPS_DATA to point to a separate directory 
    tree, which allows the Workflow and Data layers to be independently located.

* `bps.config` has local definitions which need not be propagated.  To keep it
  from being updated on git, do

```git update-index --assume-unchanged bps.config```


## Data Availability


**TODO**

Partial data available for TCGA sample TCGA-BA-4077-01 (**provide specifics**).  We do not
provide this dataset, nor the realigned data.

Provide data for steps XXX

### RNA-Seq Workflow
RNA-Seq RSEM Expression.  Optional, required for expression plot.  Can avoid steps xxx

## Usage
e.g., A_Project/1_...


## Documentation

See [BreakPointSurveyor-Core](https://github.com/ding-lab/BreakPointSurveyor-Core/README.md) 
for background.


BPS.TCGA_Virus.Lite contains a series of directories, each of which implements
a stage in the BPS workflow. The order of processing indicated by the stage's letter prefix.  
In general, a given stage may depend on data generated by a preceding stage.

Below are stages associated with the TCGA_Virus.Lite workflow and their description:

* **[A_Project](A_Project/README.md)**: Create list of BAMs, both realigned WGS and RNA-Seq.
* **[B_Reference](B_Reference/README.md)**: Reference-specific analysis and files.
* **[C_PindelRP](C_PindelRP/README.md)**: Process Pindel Read Pair (RP) breakpoint predictions.
* **[D_Discordant](D_Discordant/README.md)**: Process realigned BAM file to extract discordant human-virus reads
* **[E_Contig](E_Contig/README.md)**: Create contigs using Tigra-SV and realign them
* **[G_PlotList](G_PlotList/README.md)**: Identify target regions for further processing and visualization
* **[H_ReadDepth](H_ReadDepth/README.md)**: Evaluate read depth in target regions, obtain BAM file statistics for both WGS and RNA-Seq data
* **[I_Expression](I_Expression/README.md)**: Analyze RNA-Seq expression in vicinity of integration events.
* **[N_DrawBreakpoint](N_DrawBreakpoint/README.md)** Plot breakpoint coordinates from various predictors to breakpoint panel GGP.
* **[O_DrawDepth](O_DrawDepth/README.md)** Create read depth/copy number panel GGP and add breakpoint predictions
* **[P_DrawAnnotation](P_DrawAnnotation/README.md)** Create annotation panel GGP showing genes and exons
* **[Q_DrawHistogram](Q_DrawHistogram/README.md)**: Create histogram panel GGP showing distribution of read depth
* **[T_AssembleBPS](T_AssembleBPS/README.md)**: Assemble GGP panels into BPS structure plot and save as PDF
* **[U_RPKMBubble](U_RPKMBubble)**: Create BPS Expression plot based on expression P-values and save as PDF

### Output

* [BreakPoint Surveyor Structure Plot](T_AssembleBPS/plots/TCGA-BA-4077-01B-01D-2268-08.AA.chr14.BreakpointSurvey.pdf)
* [BreakPoint Surveyor Expression Plot](U_RPKMBubble/plots/TCGA-BA-4077-01B-01D-2268-08.AA.chr14.FDR.bubble.pdf)


## Authors
Matthew A. Wyczalkowski, m.wyczalkowski@wustl.edu

## License
This software is licensed under the GNU General Public License v3.0

## Acknowledgements
