# BreakPointSurveyor: Synthetic branch

*A comprehensive workflow to generate, analyze, and visualize interchromosomal breakpoints in a synthetic dataset.*

See [TCGA_Virus](https://github.com/ding-lab/BreakPointSurveyor) workflow for comprehensive documentation.

## Overview

The Synthetic branch generates an interchromosomal translocation by concatenating two reference segments,
then creating synthetic (simulated) reads in this region.  After realigning with BWA to a custom minimal reference,
this small BAM is analyzed and visualized with a standard BPS workflow.

A breakpoint is generated by concatenating these two segments together:
* Segment 1: chr9:130,840,000-130,850,000
* Segment 2: chr22:23,260,000-23,270,000

In addition, this workflow demonstrates gene and exon annotation functionality, including exon labels.

## Results

Structure plot below illustrates discordant reads mapping to a junction between chr 9 and 22, together with copy number
in that region.

<img src="T_PlotStructure/plots/synthetic.9-22.AA.chr22_chr9.BreakpointSurvey.png" width="600"/>

## Workflow

Stages associated with Synthetic workflow:

* **[A_Reference](A_Reference/README.md)**: Reference-specific analysis and files.
* **[B_ExonGene](B_ExonGene/README.md)**: Generate exon and gene definitions files.
* **[C_Project](C_Project/README.md)**: Create list of BAMs, both realigned WGS and RNA-Seq.  Create BAMs in `Synthetic` branch.
* **[G_Discordant](G_Discordant/README.md)**: Process realigned BAM file to extract discordant human-virus reads
* **[J_PlotList](J_PlotList/README.md)**: Identify target regions for further processing and visualization
* **[K_ReadDepth](K_ReadDepth/README.md)**: Evaluate read depth in target regions, obtain BAM file statistics for both WGS and RNA-Seq data
* **[N_DrawBreakpoint](N_DrawBreakpoint/README.md)** Plot breakpoint coordinates from various predictors to breakpoint panel GGP.
* **[O_DrawDepth](O_DrawDepth/README.md)** Create read depth/copy number panel GGP and add breakpoint predictions
* **[P_DrawAnnotation](P_DrawAnnotation/README.md)** Create annotation panel GGP showing genes and exons
* **[Q_DrawHistogram](Q_DrawHistogram/README.md)**: Create histogram panel GGP showing distribution of read depth
* **[T_PlotStructure](T_PlotStructure/README.md)**: Assemble GGP panels into BPS structure plot and save as PDF

## Installation

[Installation instructions](https://github.com/ding-lab/BreakPointSurveyor/blob/master/INSTALL.md).  Besides the core libraries, also install,
* BWA
* samtools
* Perl
  * Install required library with, `cpanm XML::XPath`

## Performance

* Downloading and processing of two-chromosome reference (A_Reference) is about 5 minutes, uses about 684M
* Total execution time for steps B-T about 25 seconds

